File: LeetCode-Programs/Intersection of Two Arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        sort(nums1.begin(), nums1.end());
        sort(nums2.begin(), nums2.end());
        int n=nums1.size();
        int m=nums2.size();
        vector<int> res;
        int i=0,j=0;
        while (i<n && j<m) {
        if (nums1[i]>nums2[j]) 
        {
            j++;
        }
        else if (nums2[j]>nums1[i]) 
        {
            i++;
        }
        else
        {
            res.push_back(nums1[i]);
            i++;
            j++;
        }
    }
        res.erase(unique(ans.begin(),ans.end()),ans.end());
        return res;
    }
};

File: LeetCode-Programs/Intersection of Two Arrays.py
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        nums3=[value for value in nums1 if value in nums2]
        nums3= list(set(nums3))
        return nums3
    
    
    
    
    
    #second
    nums1.sort()
    nums2.sort()
    n=len(nums1)
    m=len(nums2)
        
    i=0
    j=0
    res=[]
    while(i<n  and j<m):
         if(nums1[i]>nums2[j]):
             j=j+1
                
         elif (nums2[j]>nums1[i]):
              i=i+1
                
         else:
              res.append(nums1[i])
              i=i+1
              j=j+1
                
         res= list(set(res))
        return res
                    
    
    
